### ISSUE TITLE
Start the target framework in the same process (do not start a new process) [PAXRUNNER-250]

### ISSUE THREAD

**Issue by ops4j-issues** [ops4j/org.ops4j.pax.runner#278]
**[Alin Dreghiciu](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:a786db11-2013-41fb-af1d-ca5eb8f74aa0)** created **[PAXRUNNER-250](https://ops4j1.jira.com/browse/PAXRUNNER-250)** 

By default Pax Runner will start a the target platform (felix, Equinox,...) in a new process. There should be possible to start the target framework in the same process as the Pax runner itself, so no new process is created.<br /> This has the limitation that setting vmOptions is not possible beside system properties (-D).

---

**Fixed in:** 0.20.0
Votes: 0, Watches: 1

**Comment by ops4j-issues** [comment]
**[Alin Dreghiciu](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:a786db11-2013-41fb-af1d-ca5eb8f74aa0)** commented

To start target framework in the same process use the **inProcess** executor as in:



```java
pax-run --executor=inProcess
```



When `--vmOptions` is used only system properties (`-D`) are considers. For the rest an warning is issued.

**Comment by ops4j-issues** [comment]
**[Wojciech Durczyński](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:8550b079-2021-4188-aea7-2b0b3d6c2801)** commented

I've got a problem with new executor. When using it there is an error. It seems that URLClassLoader cannot find framework's main class. It happens on windows using 0.20.0-SNAPSHOT version of pax-runner.



jvm 1 | -&gt; Using execution environment [J2SE-1.6](https://ops4j1.jira.com/wiki/display/paxrunner/J2SE-1.6)<br /> jvm 1 | -&gt; Create equinox configuration ini file [runner\equinox\config.ini](https://ops4j1.jira.com/wiki/display/paxrunner/runner%5Cequinox%5Cconfig.ini)<br /> jvm 1 | -&gt; Using class org.ops4j.pax.runner.platform.InProcessJavaRunner [org.eclipse.core.runtime.adaptor.EclipseStarter](https://ops4j1.jira.com/wiki/display/paxrunner/org.eclipse.core.runtime.adaptor.EclipseStarter)<br /> jvm 1 | -&gt; VM options: [[-Dorg.osgi.service.http.port=%HTTP_PORT%, -Dbundles.configuration.location=C:\janosik-runner-1.6-SNAPSHOT-rUNKNOWNREVISION-all\admin\../config, -Dosgi.framework.extensions=org.eclipse.equinox.weaving.hook, -Dwicket.configuration=deployment, -Daj.weaving.verbose=false, -Dorg.aspectj.osgi.debug=false, -Dorg.aspectj.weaver.showWeaveInfo=false, -Dorg.ops4j.pax.web.session.timeout=%SESSION_TIMEOUT%, -Dosgi.console=%OSGI_CONSOLE_PORT%, -Dosgi.install.area=equinox]]<br /> jvm 1 | -&gt; Classpath: [[bundles/org.eclipse.osgi_3.4.0.v20080605-1900.jar, bundles/org.eclipse.equinox.weaving.hook_1.0.0.200809192119.jar](https://ops4j1.jira.com/wiki/display/paxrunner/bundles%2Forg.eclipse.osgi_3.4.0.v20080605-1900.jar%2C+bundles%2Forg.eclipse.equinox.weaving.hook_1.0.0.200809192119.jar)]<br /> jvm 1 | -&gt; Platform options: [[-configuration, equinox](https://ops4j1.jira.com/wiki/display/paxrunner/-configuration%2C+equinox)]<br /> jvm 1 | -&gt; Java home: [C:\Program Files\Java\jdk1.6.0_07](https://ops4j1.jira.com/wiki/display/C/%5CProgram+Files%5CJava%5Cjdk1.6.0_07)<br /> jvm 1 | -&gt; Working dir: [runner](https://ops4j1.jira.com/wiki/display/paxrunner/runner)<br /> jvm 1 |<br /> jvm 1 | ___<br /> jvm 1 | / /<br /> jvm 1 | / / Oops, there has been a problem!<br /> jvm 1 | / /<br /> jvm 1 | /__/ org.ops4j.pax.runner.platform.PlatformException: Cannot find target framework main class<br /> jvm 1 | ___<br /> jvm 1 | /__/<br /> jvm 1 |<br /> jvm 1 | -&gt; Exception catched during execution:<br /> jvm 1 | java.lang.RuntimeException: org.ops4j.pax.runner.platform.PlatformException: Cannot find target framework main class<br /> jvm 1 | at org.ops4j.pax.runner.Run.startPlatform(Run.java:627)<br /> jvm 1 | at org.ops4j.pax.runner.Run.start(Run.java:208)<br /> jvm 1 | at org.ops4j.pax.runner.Run.main(Run.java:137)<br /> jvm 1 | at org.ops4j.pax.runner.Run.main(Run.java:110)<br /> jvm 1 | at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br /> jvm 1 | at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)<br /> jvm 1 | at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)<br /> jvm 1 | at java.lang.reflect.Method.invoke(Unknown Source)<br /> jvm 1 | at org.tanukisoftware.wrapper.WrapperSimpleApp.run(WrapperSimpleApp.java:236)<br /> jvm 1 | at java.lang.Thread.run(Unknown Source)<br /> jvm 1 | Caused by: org.ops4j.pax.runner.platform.PlatformException: Cannot find target framework main class<br /> jvm 1 | at org.ops4j.pax.runner.platform.InProcessJavaRunner.exec(InProcessJavaRunner.java:101)<br /> jvm 1 | at org.ops4j.pax.runner.platform.internal.PlatformImpl.start(PlatformImpl.java:228)<br /> jvm 1 | at org.ops4j.pax.runner.Run.startPlatform(Run.java:623)<br /> jvm 1 | ... 9 more<br /> jvm 1 | Caused by: java.lang.ClassNotFoundException: org.eclipse.core.runtime.adaptor.EclipseStarter<br /> jvm 1 | at java.net.URLClassLoader$1.run(Unknown Source)<br /> jvm 1 | at java.security.AccessController.doPrivileged(Native Method)<br /> jvm 1 | at java.net.URLClassLoader.findClass(Unknown Source)<br /> jvm 1 | at java.lang.ClassLoader.loadClass(Unknown Source)<br /> jvm 1 | at java.lang.ClassLoader.loadClass(Unknown Source)<br /> jvm 1 | at org.ops4j.pax.runner.platform.InProcessJavaRunner.exec(InProcessJavaRunner.java:91)<br /> jvm 1 | ... 11 more

**Comment by ops4j-issues** [comment]
**[Alin Dreghiciu](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:a786db11-2013-41fb-af1d-ca5eb8f74aa0)** commented

Forgot to mention that this activator just works with absolute paths enabled ( [PAXRUNNER-248](https://ops4j1.jira.com/browse/PAXRUNNER-248) ) as in:



```java
pax-run --executor=inProcess --absoluteFilePaths
```



I will make this option implicit if `--executor` is used.

**Comment by ops4j-issues** [comment]
**[Alin Dreghiciu](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:a786db11-2013-41fb-af1d-ca5eb8f74aa0)** commented

I made `--absoluteFilePaths` implicit if `--executor=inProcess` so now it should run in your original setup.

**Comment by ops4j-issues** [comment]
**[Wojciech Durczyński](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:8550b079-2021-4188-aea7-2b0b3d6c2801)** commented

It helped but there is another problem now. Platform starts. Configuration file config.ini seems good. But org.eclipse.osgi is the only bundle which is installed. Others doesn't start at all.



Log:



jvm 1 | -&gt; Using execution environment [J2SE-1.6](https://ops4j1.jira.com/wiki/display/paxrunner/J2SE-1.6)<br /> jvm 1 | -&gt; Create equinox configuration ini file [runner\equinox\config.ini](https://ops4j1.jira.com/wiki/display/paxrunner/runner%5Cequinox%5Cconfig.ini)<br /> jvm 1 | -&gt; Using class org.ops4j.pax.runner.platform.InProcessJavaRunner [org.eclipse.core.runtime.adaptor.EclipseStarter](https://ops4j1.jira.com/wiki/display/paxrunner/org.eclipse.core.runtime.adaptor.EclipseStarter)<br /> jvm 1 | -&gt; VM options: [[-Dorg.osgi.service.http.port=48189, -Dbundles.configuration.location=C:\janosik-runner-1.6-SNAPSHOT-rUNKNOWNREVISION-all\admin\../config, -Dosgi.framework.extensions=org.eclipse.equinox.weaving.hook, -Dwicket.configuration=deployment, -Dcom.comarch.p7.jmx.rmiport=48184, -Dcom.comarch.p7.jmx.jmxport=48185, -Daj.weaving.verbose=false, -Dorg.aspectj.osgi.debug=false, -Dorg.aspectj.weaver.showWeaveInfo=false, -Dorg.ops4j.pax.web.session.timeout=30, -Dosgi.console=48187, -Dosgi.install.area=C:\janosik-runner-1.6-SNAPSHOT-rUNKNOWNREVISION-all\admin\runner\equinox]]<br /> jvm 1 | -&gt; Classpath: [[C:\janosik-runner-1.6-SNAPSHOT-rUNKNOWNREVISION-all\admin\runner\bundles\org.eclipse.osgi_3.4.0.v20080605-1900.jar, bundles/org.eclipse.equinox.weaving.hook_1.0.0.200809192119.jar]]<br /> jvm 1 | -&gt; Platform options: [[-configuration, C:\janosik-runner-1.6-SNAPSHOT-rUNKNOWNREVISION-all\admin\runner\equinox](https://ops4j1.jira.com/wiki/display/-configuration%2C+C/%5Cjanosik-runner-1.6-SNAPSHOT-rUNKNOWNREVISION-all%5Cadmin%5Crunner%5Cequinox)]<br /> jvm 1 | -&gt; Java home: [C:\Program Files\Java\jdk1.6.0_07](https://ops4j1.jira.com/wiki/display/C/%5CProgram+Files%5CJava%5Cjdk1.6.0_07)<br /> jvm 1 | -&gt; Working dir: [runner](https://ops4j1.jira.com/wiki/display/paxrunner/runner)<br /> jvm 1 | -&gt; Runner has successfully finished his job!<br /> jvm 1 |<br /> jvm 1 | Listening on port 48187 ...<br /> jvm 1 | -&gt; URLStreamHandlerFactory already set in the system. Replacing it with a composite



and OSGi console output:<br /> osgi&gt; ss



Framework is launched.



id State Bundle<br /> 0 ACTIVE org.eclipse.osgi_3.4.0.v20080605-1900



What's wrong?

**Comment by ops4j-issues** [comment]
**[Alin Dreghiciu](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:a786db11-2013-41fb-af1d-ca5eb8f74aa0)** commented

That is strange. Can you provide the generated config.ini, and the startup command line and/or arguments file you are using?

**Comment by ops4j-issues** [comment]
**[Alin Dreghiciu](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:a786db11-2013-41fb-af1d-ca5eb8f74aa0)** commented

I consider the issue at hand solved as it is about adding the inProcess executor and I'm about to release 0.20.0.<br /> For your issue please open an bug against version 0.20.0.

**Comment by ops4j-issues** [comment]
**[Alin Dreghiciu](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:a786db11-2013-41fb-af1d-ca5eb8f74aa0)** commented

Released as part of 0.2.0
